### ISSUE TITLE
Node now supports Regexp hasIndices -- typescript does not.

### ISSUE THREAD

**Issue by BobFrankston** [microsoft/TypeScript#44227]
# Bug Report

### 🔎 hasIndices

Node 16 supports indices, TypeScript does not.

### 🕗 Version & Regression Information

When I discovered the feature in Node 16

### ⏯ Playground Link

```
 /(a)/d.exec('a').indices
```

### 🙁 Actual behavior

In Node it works

WIth Typescript, it is flagged as an error

### 🙂 Expected behavior

New features. await support in ESNext.

**Comment by JoshuaWise** [comment]
I'm also needing this feature.

**Comment by nickmccurdy** [comment]
As a workaround, I've found these declarations helpful:
```ts
declare global {
  interface RegExpMatchArray {
    indices?: [number, number][] & {
      groups?: {
        [key: string]: [number, number]
      }
    }
  }

  interface RegExpExecArray {
    indices?: [number, number][] & {
      groups?: {
        [key: string]: [number, number]
      }
    }
  }

  interface RegExp {
    readonly hasIndices: boolean
  }
}
```
